### DISCUSSION TITLE
staggering per temp drive

### DISCUSSION THREAD

**Discussion by hajes** [ericaltendorf/plotman#874] [Ideas]
I have added second NVMe...I have currently two temp folders on two different drives. I have seven destination drives.

I wonder why Plotman doesn't issue two plots at once (one plot per temp drive). It simply waits/stagger before it issues command for second plot.

It makes sense per temp drive but I see no point to stagger second idling temp drive.

I didn't find anything in config. Any suggestions or is it limited functionality, please?

**Comment by altendky** [comment]
Your tmp drive isn't the only resource that gets used so it isn't the only reason to stagger.

**Reply by hajes** [reply]
resource management is up to operator ;-)

I have got 16/32 core Ryzen CPU that is used barely 80%, 64GB RAM that is used barely 60%, bored to dead -2 temp drive.

Two temp drives that are pointlessly staggered and can run independently from each other - instead they are treated as one and scheduled in serial fashion ;-)

**Reply by altendky** [reply]
I'm confused, why will it be better to start, for example, 2x plots every 10 minutes and aligned rather than 1x every 5 minutes and staggered?  Please share your complete configuration file.  There are lots of things you can do with the system already.  But, if "resource management is up to the operator" then they may need to write the rules that they want.  I have started some steps towards the pieces of plotman being more isolated and at some point even usable as a library.  Then the operator can write some Python code for scheduling however they want and leverage the underlying plot launching and monitoring code etc.

**Reply by ericaltendorf** [reply]
One of the primary functions of Plotman is resource management.  Starting multiple jobs at exactly the same time feels faster when you're just getting plotting started, but ultimately speed is about throughput, and throughput tends to be maximized when you  evenly distribute various workloads over time and amortize resources -- hence, even staggering.  I think I covered this in some of the "motivation and mode of operation" section here https://chiadecentral.com/plotman-automate-and-manage-chia-plotting/ , if not also in the plotman wikis....  There could be special cases where you want to do something different, but that's the base case we're starting from.

**Reply by hajes** [reply]
exactly, that is why I do not understand why stagger two separate temp drives ;-)

what is point of having idle temp drive and waiting preset amount of time until next job is executed....THAT IS WASTE OF RESOURCES.

Only reason you wanna stagger if you have only one destination drive

tmp1 > dst1
tmp2 > dst2  what do you save by idling 30min job that can run parallel?

**Reply by altendky** [reply]
You clearly did not read, or at least did not believe, my initial response.  You also did not share your configuration file to help us have a more concrete discussion.  If you want the conversation to continue stop acting like we are clueless and share your full configuration file and system description.

**Reply by hajes** [reply]
it seems to me that we all talk about something else ;-)

doesn't matter...I wrote my Python script to use numactl and separate temp drive handling instead staggering all together as Plotman does. It is just a suggestion, do whatever you want.

config is default - set to max 7 plots at once 1:2 or 25 minutes stagger.

AMD Ryzen 9 3950X, 64GB RAM, 1. temp Corsair MP 600 2TB, 2.temp WD SN850 1TB, temp for 3rd stage Samsung 780 EVO 2TB SSD, 8 x SATA III disks for final plots...nothing special.

**Reply by altendky** [reply]
Could you please start providing the information I ask for?  That is not your full configuration file.  You are concerned that you are wasting your tmp disks.  None of us try to do that.  plotman is not designed to waste tmp drives.  For this discussion to go anywhere useful for anyone you will need to provide more complete information.

Please share your complete configuration file.  Please share the output of `df -h`.  Please share how you are running plotman (there are multiple ways).  Please share the output of `plotman status` and also `plotman interactive --no-autostart-plotting --no-autostart-archiving`.

> it seems to me that we all talk about something else ;-)

I have tried to explain the common misunderstandings that fit with your question and comments.  I have also tried to better understand your setup and the resulting situation.  You have to fill in the information I asked for above so that we can start talking about the actual situation you are in.

**Reply by hajes** [reply]
still don't get it, waste of energy.

close the ticket and don't worry about it or read it over and over again till you understand what I am trying to say ;-)
